xbmc/platform/android/activity/XBMCApp.cpp

```diff
@@ -612,7 +612,6 @@ void CXBMCApp::SetRefreshRateCallback(CVariant* rateVariant)
 void CXBMCApp::SetDisplayModeCallback(CVariant* variant)
 {
   int mode = (*variant)["mode"].asInteger();
-  float rate = (*variant)["rate"].asFloat();
```

Comment:
If the only variant value needed now is "mode", the variant can be simplified to be just an int instead of an array. Code in `CXBMCApp::SetDisplayMode` to create the `CVariant` could be simplified to read just `CVariant* variant = new CVariant(mode);` and the reading code here to`float rate = variant->asInteger();´.

Reply:
Sloved.